src/EditorFeatures/CSharp/StringCopyPaste/KnownSourcePasteProcessor.cs

```diff
@@ -132,38 +127,169 @@ private void AppendContentForLiteral(StringBuilder builder)
                     throw ExceptionUtilities.UnexpectedValue(content.Kind);
                 }
             }
+
+            return ImmutableArray.Create(new TextChange(_selectionSpanBeforePaste, builder.ToString()));
         }
 
-        private void AppendContentForInterpolatedString(StringBuilder builder)
+        private ImmutableArray<TextChange> GetEditsForRawString(CancellationToken cancellationToken)
         {
+            // To make a change to a raw string we have to go through several passes to determine what to do.
+            //
+            // First, just take the copied text and determine the most basic edit that would insert it into the
+            // destination string.  Importantly, do not insert interpolations in this step (instead just replace them
+            // with a dummy character).
+            //
+            // Second, after this text is inserted, look at the content regions of the string after the paste and look
+            // at the sequences of `"` and `{` in them to see if we need to update the delimiters of the raw string.
+            // Note: this is why it is critical that any interpolations are not inserted.  We don't want the content of
+            // the interpolation to affect the delimiters.  e.g. a interpolation containing `""""` *inside* of it
+            // doesn't require updating the delimiters of the outer expression.
+            //
+            // Also, after the text is inserted, look to see if we need to convert a single-line raw expression to
+            // multi-line.
+            //
+            // At this point, we will now have the information necessary to actually insert the content and do things 
+            // like give interpolations the proper number of braces for the final string we're making.
+
+            var dummyContentEdit = GetContentEditForRawString(insertInterpolations: false, dollarSignCount: -1, indentationWhitespace: "");
```

Comment:
this method is very long, but is described by the above comment.  i intend to break it apart into pieces to make it clearer as to what happens.